src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```
@@ -12,11 +15,33 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+                throw new DataProcessingException(
+                        "Can't insert actor= " + actor + ", error= " + e);
+            }
```

This violates the checklist item: "Remember to add catch blocks for all DAO methods and throw DataProcessingException in all cases, not only if transaction != null." If transaction is null, the exception will be swallowed. You should always throw DataProcessingException in the catch block, regardless of transaction state.